test/Microsoft.Health.Dicom.Web.Tests.E2E/Functions/HostJsonFileConfigurationSource.cs

```diff
@@ -0,0 +1,198 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using EnsureThat;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Web.Tests.E2E.Functions
+{
+    // This class is a subset of the same class found inside of the Azure Functions Host:
+    // https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs
+    public class HostJsonFileConfigurationSource : IConfigurationSource
+    {
+        public string ScriptPath { get; }
+
+        public HostJsonFileConfigurationSource(string scriptPath)
+            => ScriptPath = EnsureArg.IsNotEmpty(scriptPath, nameof(scriptPath));
```

Comment:
IsNotEmptyOrWhiteSpace?

Reply:
Actually, I think I'll only check `null` as the "current" directory would be empty string and that would be fine